rs/tests/networking/canister_http_non_replicated_test.rs

```diff
@@ -0,0 +1,172 @@
+/* tag::catalog[]
+Title:: Basic HTTP requests from canisters
+
+Goal:: Ensure simple HTTP requests can be made from canisters.
+
+Run with:
+ict test //rs/tests/networking:canister_http_non_replicated_test -- --test_tmpdir=./canister_http_non_replicated_test
+
+Runbook::
+0. Instantiate a universal VM with a webserver
+1. Instantiate an IC with one application subnet with the HTTP feature enabled.
+2. Install NNS canisters
+3. Install the proxy canister
+4. Make an update call to the proxy canister.
+
+Success::
+1. Received http response with status 200.
+
+end::catalog[] */
+
+use anyhow::bail;
+use anyhow::Result;
+use canister_http::*;
+use canister_test::Canister;
+use dfn_candid::candid_one;
+use ic_cdk::api::call::RejectionCode;
+use ic_management_canister_types_private::{HttpMethod, TransformContext, TransformFunc};
+use ic_system_test_driver::driver::group::SystemTestGroup;
+use ic_system_test_driver::driver::{
+    test_env::TestEnv,
+    test_env_api::{READY_WAIT_TIMEOUT, RETRY_BACKOFF},
+};
+use ic_system_test_driver::systest;
+use ic_system_test_driver::util::block_on;
+use proxy_canister::UnvalidatedCanisterHttpRequestArgs;
+use proxy_canister::{RemoteHttpRequest, RemoteHttpResponse};
+use slog::{info, Logger};
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(canister_http::setup)
+        .add_test(systest!(test))
```

Comment:
nit: can we split this into two test cases?

```suggestion
        .add_test(systest!(test_non_replicated_random_endpoint_works))
        .add_test(systest!(test_fully_replicated_random_endpoint_fails))
```

Reply:
I think we should keep it as a single test for now; they would both need to install the canister etc. Alternatively we could move the canister installation in the setup, but that would upset other tests... I'll leave this like this for now and try to improve it later